### PR TITLE
feat: on UI side filtering put typed sign V4 requests for which decoding data is displayed

### PR DESCRIPTION
**Pull request by jpuri**
## **Description**

Filter our typed sign requests for which decoding api results are displayed.

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/3682

## **Manual testing steps**

1. Enable signature decoding locally
2. Submit request for permit signatures on test dapp of supported seaport signatures
3. It should work as expected

## **Screenshots/Recordings**
Screenshot 2024-11-27 at 3 14 40 PM
Screenshot 2024-11-27 at 3 14 31 PM

## **Pre-merge author checklist**

- [X] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [X] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by matthewwalsh0 — CHANGES_REQUESTED** [review]

File: ui/pages/confirmations/components/confirm/info/typed-sign/typed-sign-v4-simulation/typed-sign-v4-simulation.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    isNonPermitRequestSupportedByDecodingAPI(currentConfirmation) || isPermit;
+
+  if (!supportedByDecodingAPI) {
+    return null;
+  }
+
+  const { decodingData, decodingLoading } = currentConfirmation;
+
+  if (
+    ((!decodingLoading && decodingData === undefined) || decodingData?.error) &&
```
[the remark this answers is not in the document]
Comment (jpuri) [reply ui/pages/confirmations/components/confirm/info/typed-sign/typed-sign-v4-simulation/typed-sign-v4-simulation.tsx:53]: Nopes

File: ui/pages/confirmations/components/confirm/info/typed-sign/typed-sign-v4-simulation/typed-sign-v4-simulation.test.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        mockStore,
+      );
+
+      expect(await findByText('Estimated changes')).toBeInTheDocument();
+      expect(await findByText('Spending cap')).toBeInTheDocument();
+      expect(await findByText('1,461,501,637,3...')).toBeInTheDocument();
+    });
+  });
+
+  it.only('should render decoding simulation for seaport request', async () => {
```
[the remark this answers is not in the document]
Comment (jpuri) [reply ui/pages/confirmations/components/confirm/info/typed-sign/typed-sign-v4-simulation/typed-sign-v4-simulation.test.tsx:146]: yep fixed

File: ui/pages/confirmations/components/confirm/info/typed-sign/typed-sign-v4-simulation/typed-sign-v4-simulation.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (jpuri) [reply ui/pages/confirmations/components/confirm/info/typed-sign/typed-sign-v4-simulation/typed-sign-v4-simulation.tsx:53]: I updated condition to `decodingLoading === undefined && decodingData === undefined`. Id none of the fields is present it implies that decoding api call is not made for the request and so we can return null.

File: ui/pages/confirmations/components/confirm/info/typed-sign/typed-sign-v4-simulation/typed-sign-v4-simulation.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    primaryTypeList: ['BulkOrder'],
+    versionList: ['1.4', '1.5', '1.6'],
+  },
+  {
+    domainName: 'Seaport',
+    primaryTypeList: ['OrderComponents'],
+  },
+];
+
+const isNonPermitRequestSupportedByDecodingAPI = (
```
[the remark this answers is not in the document]
Comment (jpuri) [reply ui/pages/confirmations/components/confirm/info/typed-sign/typed-sign-v4-simulation/typed-sign-v4-simulation.tsx]: fixed

**Review by matthewwalsh0 — APPROVED** [review]

**Review by pedronfigueiredo — APPROVED** [review]

**Pull request merged**